### PR TITLE
Introduce AnimationValue to manage animated properties in Gecko.

### PR DESCRIPTION
**Pull request by emilio**
Doesn't compile because there are a bunch of `from_computed_value` missing, and I need to take care of fixing that before, but open for discussion.

@birtles: is an interface like this something like what you were looking for?

cc @Manishearth

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13694)

### CODE REVIEW

**Comment by birtles** [comment]
Looks great. Ultimately we'll want to more than just interpolate (we'll want to do distance, addition etc. too) but for now this looks great.
One minor comment is that for interpolating the interpolation parameter should probably be named 'progress' instead of 'time' since it doesn't really represent time but I suppose it's named time to match other parts of the code.

File: components/style/properties/helpers/animated_properties.mako.rs [hunk 1]
```diff
@@ -144,6 +144,64 @@ impl AnimatedProperty {
...
+            % for prop in data.longhands:
+                % if prop.animatable:
+                    (&AnimationValue::${prop.camel_case}(ref from),
+                     &AnimationValue::${prop.camel_case}(ref to)) => {
+                        from.interpolate(to, time).map(AnimationValue::${prop.camel_case})
+                    }
+                % endif
+            % endfor
+            _ => {
+                panic!("Expected interpolation of computed values of the same \
```
Comment (emilio) [inline components/style/properties/helpers/animated_properties.mako.rs]: Nope, I meant to panic, that's a programming error and not an interpolation error. We can make it throw an error if we expect that, but I think panicking if interpolating different properties is what we should do in the short term at least.

**Comment by Manishearth** [comment]
r=me if it compiles

**Comment by emilio** [comment]
Now it does.

@bors-servo: r=Manishearth,birtles

**Pull request merged**